engine/apps/alerts/escalation_snapshot/utils.py

```diff
@@ -1,29 +1,33 @@
+import datetime
+import typing
+
 import pytz
 from django.utils import timezone
 
 
-def eta_for_escalation_step_notify_if_time(from_time, to_time, current_time=None) -> timezone.datetime:
+def eta_for_escalation_step_notify_if_time(
+    from_time: datetime.time, to_time: datetime.time
+) -> typing.Optional[datetime.datetime]:
     """
     Counts eta for STEP_NOTIFY_IF_TIME
-    :return: timezone.datetime
     """
-    eta = current_time
-    current_time = current_time or timezone.now()
+    eta: typing.Optional[datetime.datetime] = None
+
+    now = timezone.now()
+    current_date = now.date()
+    current_time = now.time()
+
     if from_time < to_time:
-        if from_time > current_time.time():
-            eta = timezone.datetime.combine(current_time.date(), from_time).astimezone(pytz.UTC)
-        elif current_time.time() >= to_time:
-            eta = timezone.datetime.combine((current_time + timezone.timedelta(days=1)).date(), from_time).astimezone(
```

Comment:
`current_time` was always previously equal to `timezone.now()` since the `current_time` arg passed into the function was always `None`, hence changing `current_time` to `now` in this line